benches/benchmark.rs

```diff
@@ -0,0 +1,440 @@
+#[macro_use]
+extern crate criterion;
+
+use criterion::Criterion;
+use graphlib::*;
+
+// includes benches for :
+// 1. new() -> Graph<T>
+// 2. with_capacity(capacity: usize) -> Graph<T>
+fn bench_create(c: &mut Criterion) {
+    c.bench_function("new", |b| b.iter(|| Graph::<usize>::new()));
+    c.bench_function("with_capacity", |b| {
+        b.iter(|| Graph::<usize>::with_capacity(7))
+    });
+}
+
+// includes benches for :
+// 1. add_edge(&mut self, a: &VertexId, b: &VertexId) -> Result<(), GraphErr>
+// 2. add_vertex(&mut self, item: T) -> VertexId
+// 3. capacity(&self) -> usize
+// 4. edge_count(&self) -> usize
+// 5. fetch(&self, id: &VertexId) -> Option<&T>
+// 6. fetch_mut(&mut self, id: &VertexId) -> Option<&mut T>
+// 7. fold<A>(&self, initial: A, fun: impl Fn(&T, A) -> A) -> A
+// 8. has_edge(&self, a: &VertexId, b: &VertexId) -> bool
+// 9. is_cyclic(&self) -> bool
+// 10.remove(&mut self, id: &VertexId)
+// 11.remove_edge(&mut self, a: &VertexId, b: &VertexId)
+// 12.reserve(&mut self, additional: usize)
+// 13.retain(&mut self, fun: impl Fn(&T) -> bool)
+// 14.roots_count(&self) -> usize
+// 15.shrink_to_fit(&mut self)
+// 16.vertex_count(&self) -> usize
+fn bench_others(c: &mut Criterion) {
+    c.bench_function("add_edge", |b| {
+        b.iter(|| {
```

Comment:
Can you separate the setup code from the benchmarked code like so:

```rust
c.bench_function("add_edge", |b| {
    let mut graph: Graph<usize> = Graph::new();

    let v1 = graph.add_vertex(1);
    let v2 = graph.add_vertex(2);
    
    b.iter(|| {
        graph.add_edge(&v1, &v2);
    })
});
```

Reply:
my bad